pkg/driver/driver_openstack.go

```diff
@@ -276,6 +320,11 @@ func (d *OpenStackDriver) Delete(machineID string) error {
 	} else if len(res) == 0 {
 		// No running instance exists with the given machine-ID
 		klog.V(2).Infof("No VM matching the machine-ID found on the provider %q", machineID)
+		err = d.deletePort()
+		if err != nil {
+			return err
+		}
+
```

Comment:
Another suggestion is to revert the conditional:

```go
if len(res) > 0 {
<do server delete>
}

deletePort()
```

Because it seems that you need to call deletePort in all cases anyway (whether or not the machine was found)

Reply:
True, calling deletePort() twice is ugly. Changed the function to always call deletePort() as the last step and put the server delete login into the conditional as suggested.